controller/client/client.go

```diff
@@ -66,77 +90,220 @@ func normalizeConfig(conf *Config) error {
 	return nil
 }
 
-type client struct {
-	url        string
-	authToken  string
-	httpClient *http.Client
+// ListRules returns the list of rules
+func (c *Client) ListRules(filter *api.RuleFilter) (*api.RulesSet, error) {
+	var rulesSet api.RulesSet
+
+	path := rulesPath
+	query := c.filterToStringQuery(filter)
```

Comment:
Why not use the default library for building the URL?:
https://golang.org/pkg/net/url/#example_URL